test/OpenTelemetry.Exporter.Prometheus.Tests/PrometheusSerializerTests.cs

```diff
@@ -383,5 +384,290 @@ public void HistogramNaN()
+                .AddInMemoryExporter(metrics)
+                .Build();
+
+            var counter = meter.CreateCounter<long>("test_counter");
+            counter.Add(10, new KeyValuePair<string, object>("key", value));
+
+            provider.ForceFlush();
+
+            var cursor = PrometheusSerializer.WriteMetric(buffer, 0, metrics[0]);
+
+            var labelString = string.Empty;
+            switch (value)
+            {
+                case Array array:
+                    var expectedArray = new double[array.Length];
+                    for (var i = 0; i < array.Length; i++)
+                    {
+                        expectedArray[i] = Convert.ToInt64(array.GetValue(i));
+                    }
+
+                    labelString = $"{{key='\\[{string.Join(",", expectedArray)}\\]'}}";
+                    break;
+                default:
+                    labelString = $"{{key='{Convert.ToInt64(value)}'}}";
+                    break;
+            }
+
+            Assert.Matches(
+                ("^"
+                    + "# TYPE test_counter counter\n"
+                    + $"test_counter{labelString} 10 \\d+\n"
+                    + "$").Replace('\'', '"'),
+                Encoding.UTF8.GetString(buffer, 0, cursor));
+        }
+
+        [Theory]
+        [InlineData(float.MaxValue)]
+        [InlineData(double.MaxValue)]
+        [InlineData(new float[] { 1, 2, 3 })]
+        [InlineData(new double[] { 1, 2, 3 })]
+        public void FloatingPointTypesSupported(object value)
+        {
+            var buffer = new byte[85000];
+            var metrics = new List<Metric>();
+
+            using var meter = new Meter(Utils.GetCurrentMethodName());
+            using var provider = Sdk.CreateMeterProviderBuilder()
+                .AddMeter(meter.Name)
+                .AddInMemoryExporter(metrics)
+                .Build();
+
+            var counter = meter.CreateCounter<double>("test_counter");
+            counter.Add(10, new KeyValuePair<string, object>("key", value));
+
+            provider.ForceFlush();
+
+            var cursor = PrometheusSerializer.WriteMetric(buffer, 0, metrics[0]);
+
+            var labelString = string.Empty;
+            switch (value)
+            {
+                case Array array:
+                    var expectedArray = new double[array.Length];
+                    for (var i = 0; i < array.Length; i++)
+                    {
+                        expectedArray[i] = Convert.ToDouble(array.GetValue(i));
+                    }
+
+                    labelString = $"{{key='\\[{string.Join(",", expectedArray)}\\]'}}".Replace("+", "\\+");
+                    break;
+                default:
+                    // TODO: Verify if scientific notion is ok for Prometheus
+                    // {key="1.7976931348623157E+308"}
+                    labelString = $"{{key='{Convert.ToDouble(value)}'}}".Replace("+", "\\+");
+                    break;
```

Comment:
Is scientific notions for the string representations of double-valued attributes ok?

Reply:
Yes.
The official spec has an example https://github.com/prometheus/docs/blob/main/content/docs/instrumenting/exposition_formats.md#text-format-example:

```text
rpc_duration_seconds_sum 1.7560473e+07
```